src/commons/navigationBar/NavigationBar.tsx

```diff
@@ -65,277 +55,222 @@ const NavigationBar: React.FC = () => {
 
+  const createMobileNavlink: CreateNavlinkFunction = useCallback(
+    navbarEntry => (
       <NavLink
+        to={navbarEntry.to}
         className={({ isActive }) =>
-          classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL, {
-            [Classes.ACTIVE]: isActive
-          })
+          classNames(
+            'NavigationBar__link__mobile',
+            Classes.BUTTON,
+            Classes.MINIMAL,
+            Classes.LARGE,
+            {
+              [Classes.ACTIVE]: isActive
+            }
+          )
         }
-        to={courseId == null ? '/welcome' : `/courses/${courseId}`}
+        onClick={() => setMobileSideMenuOpen(false)}
+        key={navbarEntry.text}
       >
-        <Icon icon={IconNames.SYMBOL_DIAMOND} />
-        <NavbarHeading style={{ paddingBottom: '0px' }}>
-          {courseShortName || 'Source Academy @ NUS'}
-        </NavbarHeading>
+        <Icon icon={navbarEntry.icon} />
+        <div>{navbarEntry.text}</div>
+        {navbarEntry.hasNotifications && (
+          <NotificationBadge
+            notificationFilter={filterNotificationsByType(navbarEntry.text)}
+            disableHover={true}
+          />
+        )}
       </NavLink>
-
-      {(role || Constants.enableGitHubAssessments) && (
-        <NavigationBarMobileSideMenu
-          assessmentTypes={assessmentTypes}
-          isOpen={mobileSideMenuOpen}
-          onClose={() => setMobileSideMenuOpen(false)}
-        />
-      )}
-    </NavbarGroup>
+    ),
+    [setMobileSideMenuOpen]
   );
 
-  const desktopNavbarLeftPopoverContent = (
-    <Navbar>
-      <NavbarGroup>
-        {assessmentTypes?.map((assessmentType, idx) => (
-          <NavLink
-            to={`/courses/${courseId}/${assessmentTypeLink(assessmentType)}`}
-            className={({ isActive }) =>
-              classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL, {
-                [Classes.ACTIVE]: isActive
-              })
-            }
-            key={assessmentType}
+  const wrapWithMobileHamburger = (navlinks: (React.ReactFragment | null)[]) => {
+    // Don't render drawer when there are 0 navlinks in it
+    const nonNullNavlinks = navlinks.filter(e => e !== null);
+    const renderDrawer = nonNullNavlinks.length > 0;
+
+    return (
+      <NavbarGroup align={Alignment.LEFT}>
+        {renderDrawer && (
+          <Button
+            onClick={() => setMobileSideMenuOpen(!mobileSideMenuOpen)}
+            icon={IconNames.MENU}
+            large={true}
+            minimal={true}
+          />
+        )}
+        <NavLink
+          className={classNames('NavigationBar__link', Classes.BUTTON, Classes.MINIMAL)}
+          to={
+            Constants.playgroundOnly ? '/' : courseId == null ? '/welcome' : `/courses/${courseId}`
+          }
+        >
+          <Icon icon={IconNames.SYMBOL_DIAMOND} />
+          <NavbarHeading style={{ paddingBottom: '0px' }}>
+            {courseShortName || 'Source Academy'}
```

Comment:
Can we define a default fallback value in `Constants.ts` instead of hardcoding it here? It would make it easier for not only customisation, but also i18n efforts.